frontend/src/reducers/ModuleEditorContextReducer.ts

```diff
@@ -6,10 +6,21 @@ import {
   LessonType,
   ContentBlock,
   ContentTypeEnum,
+  EditorChangeStatus,
+  EditorChangeStatuses,
 } from "../types/ModuleEditorTypes";
 
 /* eslint-disable no-console */
 
+const updateChangeStatus = (
+  hasChanged: EditorChangeStatuses,
+  docID: string,
+  status: EditorChangeStatus,
+): EditorChangeStatuses => {
+  if (!(docID in hasChanged)) return { ...hasChanged, [docID]: status };
```

Comment:
I believe this logic breaks for DELETE actions; if the user performs a CREATE or UPDATE first, then a DELETE, the final state of the system should be that the document is deleted, but based on this logic, that won't happen. I think we probably need some special cases here; maybe we can leave a TODO for when we add the ability to delete lessons individually?